chromium_src/net/base/lookup_string_in_fixed_set.cc

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#define LookupSuffixInReversedSet LookupSuffixInReversedSet_ChromiumImpl
+#include "../../../../net/base/lookup_string_in_fixed_set.cc"
+#undef LookupSuffixInReversedSet
+
+#include "base/strings/string_util.h"
+
+namespace net {
+
+int LookupSuffixInReversedSet(const unsigned char* graph,
```

Comment:
Here's how chromium handles public suffix list, updates in the upstream file is pulled into https://source.chromium.org/chromium/chromium/src/+/master:net/base/registry_controlled_domains/effective_tld_names.dat and they will run a script to generate the corresponding change in effective_tld_names.gperf and push both updates in the same commit.
When building chromium, `net/tools/dafsa/make_dafsa.py` will be run which generate a byte array in `effective_tld_names-reversed-inc.cc` from the list to be used for comparison in run-time.
This function is the only function for looking up entries in the public suffix list, so we hook into this function to treat ipfs/ipns.localhost as a match too.

Reply:
Can we put a short summary of this explanation into the code?

Reply:
Added in https://github.com/brave/brave-core/pull/7046/commits/267677bf21dc69473552a23845871754669082f6.